### ISSUE TITLE
docker-compose up converts linux path to windows for volumes

### ISSUE THREAD

**Issue by AlekseiPavlov-GC** [docker/compose#8051]
Hi colleagues,

**Steps to reproduce**

Docker 3.1.0 (51484) running with WSL2 on Windows 10 version 1909.

I try to execute the script using Windows Powershell 7.1:
`docker-compose -f "docker-compose.generated.json" up --build  --remove-orphans --renew-anon-volumes --force-recreate  --always-recreate-deps`

The json file has the following section:
```
"volumes": {
       "inputansibleconfigvolume": {
          "driver_opts": {
             "type": "none",
             "device": "${ANSIBLE_CONFIGS}",
             "o": "bind"
          }
       }
    }
```

The Environment Variable _ANSIBLE_CONFIGS_ is set to: _/c/Dev/Git/folder_

**When I execute the command:**
`docker-compose -f "docker-compose.generated.json" up --build  --remove-orphans --renew-anon-volumes --force-recreate  --always-recreate-deps`

A docker volume is created.

**Actual:**
device: "C:\\c\\Dev\\Git\\folder"
Or depending on different manipulations device value can be a slightly different but always in Windows style.

**Expected:**
device: "/c/Dev/Git/folder"

**I tried:**
COMPOSE_CONVERT_WINDOWS_PATHS=1
COMPOSE_CONVERT_WINDOWS_PATHS=0

This had no effect.

**Workaround:**
```
"volumes": {
       "inputansibleconfigvolume": {
          "external" true
       }
    }
```

`docker volume create --name inputansibleconfigvolume -o type=none -o device=/c/Dev/Git/folder -o o=bind`

Could you please check this?

**Comment by LaXiS96** [comment]
Hi, I am pretty sure the cause for this issue is the same as this one: https://github.com/docker/compose/issues/7856#issuecomment-884710005

TLDR, in commit https://github.com/docker/compose/commit/8c81a9da7a846c1f0b1fe3b40e7c7ab2e70c2215 relative path handling was added to driver_opts.device. I don't know how this change was tested, but the simple use case of running docker-compose on Windows to deploy on a Linux remote machine with a bind mounted volume, breaks.
The problem is that the path is run through `os.path.abspath` on the local machine, which in this case converts a Linux path to Windows separators and adds `C:`.
I'm still waiting for any input from @aiordache. The quickest solution would be to revert to not handling relative paths, or better handle them on the Docker daemon host instead of the docker-compose client.
